anki/template/template.py

```diff
@@ -200,20 +205,27 @@ def render_unescaped(self, tag_name=None, context=None):
 
     def clozeText(self, txt, ord, type):
         reg = clozeReg
-        if not re.search(reg%ord, txt):
+        if not re.search(reg % ord, txt):
             return ""
+
         def repl(m):
             # replace chosen cloze with type
+            noSound = removeClozeSound(m.group(3))
```

Comment:
I'm personally not convinced that this is a good idea "generally" (if it is relevant to show a hint, then it seems relevant to show it completely).
Someone with a note composed entirely of sound clozes (trying to learn a part of a song, for instance) would therefore have an empty question. I'm not saying I'd use that, but that sounds legit and your change prevents it.
In the end, that feels a bit more like an add-on feature.